reconcile/osd_mirrors_data_updater.py

```diff
@@ -0,0 +1,148 @@
+from reconcile import mr_client_gateway
+from reconcile.utils.mr import CSInstallConfig
+
+from reconcile import queries
+from reconcile.utils.aws_api import AWSApi
+
+
+QONTRACT_INTEGRATION = 'osd-mirrors-data-updater'
+LOG = logging.getLogger(__name__)
+
+
+def get_tf_resource_info(namespace, identifier):
+    """
+    Takes a namespace from app-interface and searches
+    a given ECR by its identifier.
+    """
+    tf_resources = namespace['terraformResources']
+    for tf_resource in tf_resources:
+        if 'identifier' not in tf_resource:
+            continue
+
+        if tf_resource['identifier'] != identifier:
+            continue
+
+        if tf_resource['provider'] != 'ecr':
+            continue
+
+        return tf_resource
+
+
+def get_aws_account_info(account):
+    """
+    Gets all AWS accounts from app-interface and searches the
+    desired one.
+    """
+    for account_info in queries.get_aws_accounts():
+        if 'name' not in account_info:
+            continue
+        if account_info['name'] != account:
+            continue
+        return account_info
+
+
+def get_image_uri(aws_cli, account, repository):
+    """
+    Finds the repository URL for a given ECR resource.
+    """
+    for repo in aws_cli.resources[account]['ecr']:
+        if repo['repositoryName'] == repository:
+            return repo['repositoryUri']
+
+
+def run(dry_run, gitlab_project_id=None):
+    settings = queries.get_app_interface_settings()
+    namespaces = queries.get_namespaces()
+
+    # This is a list of app-interface ECR resources and their
+    # mirrors
+    osd_mirrors = []
+    for namespace in namespaces:
+        # We are only interested on the ECR resources from
+        # this specific namespace
+        if namespace['name'] != 'osd-operators-ecr-mirrors':
+            continue
+
+        if namespace['terraformResources'] is None:
+            continue
+
+        for tfr in namespace['terraformResources']:
+            if tfr['provider'] != 'ecr':
+                continue
+            if tfr['mirror'] is None:
+                continue
+
+            osd_mirrors.append(tfr)
+
+    # Now the tricky part. The "OCP Release ECR Mirror" is a stand-alone
+    # object in app-interface. We have to process it so we get the
+    # upstream and the mirror repositories
+    instances = queries.get_ocp_release_ecr_mirror()
+    for instance in instances:
+        namespace = instance['ecrResourcesNamespace']
+        ocp_release_identifier = instance['ocpReleaseEcrIdentifier']
+        ocp_art_dev_identifier = instance['ocpArtDevEcrIdentifier']
+
+        ocp_release_tf_info = get_tf_resource_info(namespace,
+                                                   ocp_release_identifier)
+
+        # We get an ECR resource from app-interface, but it has
+        # no mirror since the mirroring is done differently there.
+        # Anyway, let's just inject the mirror information so we process
+        # all the ECR resources the same way
```

Comment:
So we just know this mirror exists? This comment is a bit daunting, you may want to really explain how things work or a reference about how to understand it 😅